docs/components/rule.md

```diff
@@ -202,6 +204,35 @@ In case of Ruler in HA you need to make sure you have the following labelling se
 
 Advanced relabelling configuration is possible with the `--alert.relabel-config` and `--alert.relabel-config-file` flags. The configuration format is identical to the [`alert_relabel_configs`](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alert_relabel_configs) field of Prometheus. Note that Thanos Ruler drops the labels listed in `--alert.label-drop` before alert relabelling.
 
+## Stateless Ruler via Remote Write
+
+Stateless ruler enables nearly indefinite horizontal scalability. Ruler doesn't have a fully functional TSDB for storing evaluation results, but uses a WAL only storage and sends data to some remote storage via remote write.
+
+The WAL only storage reuses the upstream [Prometheus agent](https://prometheus.io/blog/2021/11/16/agent/) and it is compatible with the old TSDB data. For more design purpose of this mode, please refer to the [proposal](https://thanos.io/tip/proposals-accepted/202005-scalable-rule-storage.md/).
+
+Stateless mode can be enabled by providing `--remote-write.config` or `--remote-write.config-file` flag. For example:
+
+```bash
+thanos rule \
+    --data-dir                  "/path/to/data" \
+    --eval-interval             "30s" \
+    --rule-file                 "/path/to/rules/*.rules.yaml" \
+    --alert.query-url           "http://0.0.0.0:9090" \ # This tells what query URL to link to in UI.
+    --alertmanagers.url         "http://alert.thanos.io" \
+    --query                     "query.example.org" \
+    --query                     "query2.example.org" \
+    --objstore.config-file      "bucket.yml" \
+    --label                     'monitor_cluster="cluster1"' \
+    --label                     'replica="A"' \
+    --remote-write.config-file  'rw-config.yaml'
+```
+
+The remote write config file is exactly the same as the [Prometheus remote write config format](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write).
+
+**NOTE:**
+1. Remote write metadata config is not supported in this mode and will be ignored if provided in the remote write configuration.
```

Comment:
What's "remote write metadata config"? :thinking:

Reply:
![image](https://user-images.githubusercontent.com/25150124/142905409-e1afe0ff-b3e0-4572-a40c-506bd0145c5b.png)

Metadata config depends on scrape manager so it is not used.

Reply:
Ahh, I see. Maybe we can then replace "remote write metadata config" with "remote write `metadata_config`" to emphasize that it is a key's name?

Reply:
Done